front-end/src/components/BaseButton.jsx

```diff
@@ -2,12 +2,22 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 const BaseButton = (props) => {
+    // Change button appearance based on whether it is enabled or disabled
+    const style = () => {
+        if (props.isDisabled) {
+            return "bg-blue-300 rounded-lg shadow-lg opacity-50 py-3 px-6";
+        } else {
+            return "bg-blue-300 rounded-lg shadow-lg hover:bg-blue-200 active:bg-blue-400 py-3 px-6";
+        }
```

Comment:
A rule I like (and which would minimize code duplication) is that most functions should only have 1 point of return.
It might be better to format this like:
```suggestion
        const baseStyling = "bg-blue-300 rounded-lg shadow-lg";
        const enabled = "opacity-50 py-3 px-6";
        const disabled = "hover:bg-blue-200 active:bg-blue-400 py-3 px-6";
        // ternary that appends the important parts of your if-statement
        return styling + props.isDisabled ? disabled : enabled;
```